main/streams/plain_wrapper.c

```diff
@@ -242,18 +245,19 @@ PHPAPI php_stream *_php_stream_fopen_tmpfile(int dummy STREAMS_DC)
 	return php_stream_fopen_temporary_file(NULL, "php", NULL);
 }
 
-static void detect_is_pipe(php_stdio_stream_data *self) {
+static void detect_is_seekable(php_stdio_stream_data *self) {
 #if defined(S_ISFIFO) && defined(S_ISCHR)
 	if (self->fd >= 0 && do_fstat(self, 0) == 0) {
-		self->is_pipe = S_ISFIFO(self->sb.st_mode) || S_ISCHR(self->sb.st_mode);
+		self->is_seekable = !(S_ISFIFO(self->sb.st_mode) || S_ISCHR(self->sb.st_mode));
```

Comment:
Seems the `is_pipe` not set here.

Reply:
In general, this should be crossplatform. Perhaps preserve the `self->is_pipe = (do_fstat(self, 0) == 0 && S_ISFIFO(self->sb.st_mode)) ? 1 : 0;` as it was before?

Reply:
Ah, should definitely be set for non Windows platforms as well. I don't think the `? 1 : 0` is necessary, since `is_pipe` is `unsigned:1`.